sos/plugins/logs.py

```diff
@@ -50,91 +39,41 @@ def setup(self):
             if os.path.isfile(i):
                 self.add_copy_spec(i)
 
-        if self.get_option('all_logs'):
-            self.add_journal(boot="this", since=self.get_option("since"),
-                             allfields=True, output="verbose")
-            self.add_journal(boot="last", allfields=True, output="verbose")
-
-    def postproc(self):
-        self.do_path_regex_sub(
-            r"/etc/rsyslog*",
-            r"ActionLibdbiPassword (.*)",
-            r"ActionLibdbiPassword [********]"
-        )
-        self.do_path_regex_sub(
-            r"/etc/rsyslog*",
-            r"pwd=.*",
-            r"pwd=[******]"
-        )
-
-
-class RedHatLogs(Logs, RedHatPlugin):
-
-    option_list = [
-        ("log_days", "the number of days logs to collect", "", 3)
-    ]
-
-    def setup(self):
-        super(RedHatLogs, self).setup()
-        # NOTE: for historical reasons the 'messages' and 'secure' log
-        # paths for the RedHatLogs plugin do not obey the normal --all-logs
-        # convention. The rotated copies are collected unconditionally
-        # due to their frequent importance in diagnosing problems.
-        messages = "/var/log/messages"
-        secure = "/var/log/secure"
-
-        have_messages = os.path.exists(messages)
-
-        self.add_copy_spec(secure + "*")
-        self.add_copy_spec(messages + "*")
-
-        # collect three days worth of logs by default if the system is
-        # configured to use the journal and not /var/log/messages
-        if not have_messages and self.is_installed("systemd"):
-            try:
-                days = int(self.get_option("log_days"))
-            except ValueError:
-                days = 3
-            if self.get_option("all_logs") and self.get_options("since"):
-                since = self.get_options("since")
-            else:
-                since = "-%ddays" % days
-            self.add_journal(since=since)
-
-
-class DebianLogs(Logs, DebianPlugin, UbuntuPlugin):
-
-    option_list = [
-        ("log_days", "the number of days logs to collect", "", 7)
-    ]
-
-    def setup(self):
-        super(DebianLogs, self).setup()
-        days = int(self.get_option("log_days"))
-        journal = os.path.exists("/var/log/journal/")
         self.add_copy_spec([
+            "/etc/syslog.conf",
+            "/etc/rsyslog.conf",
+            "/etc/rsyslog.d",
+            "/var/log/boot.log",
+            "/var/log/installer",
+            "/var/log/unattended-upgrades",
+            "/var/log/messages*",
+            "/var/log/secure*",
             "/var/log/udev",
             "/var/log/dist-upgrade",
-            "/var/log/installer",
-            "/var/log/unattended-upgrades"
-            ])
+        ])
+
+        self.add_cmd_output("journalctl --disk-usage")
         self.add_cmd_output('ls -alRh /var/log/')
 
+        journal = os.path.exists("/var/log/journal/")
         if journal and self.is_installed("systemd"):
+            self.add_journal()
```

Comment:
Since is now covered by the _init_ change.

Reply:
Yeah, I realised that belatedly - that kind of API change with broad impact is absolutely the kind of thing that needs to be in a separate clearly labelled commit:

```
[Plugin] automatically apply since= in add_journal()
```

It's really easy to miss a subtle point like that in a large commit that is re-shuffling a lot of code between classes in one file.

I am not sure right now that this change is OK because I don't know for certain how this will interact with other options that we pass to `journalctl` - it needs to be tested so that we don't end up with any data collection regressions.

Back when I added `add_journal()` I plumbed all the options through as kwargs because we had a request from Red Hat support that they _always_ needed the current and previous boot logs - my worry here is that using `--since` will override `--boot=last` and exclude those log messages if they fall outside the date range specified with since. @TurboTurtle did a test here and it seems to confirm that `--since` does take precedence over `--boot`, meaning that you get no messages if they are excluded by the since date.

Reply:
That's actually one of the reason I thought it needed a bigger rewrite (the bigger since change is what came from the other PR).  

The plugin as it is is quite inconsistent on this - it currently specified since for this boot.  Which means we *might* be in scenarios where using since causes us to collect a full last boot, but two incomplete copies of this boot    We can also be in the scenario where `this` boot and the main journal collection are completely equivalent (as both hit the default size limit - although technically `this` boot would likely be larger due to catalog). 

I'm happy to move the all journal since changes out of this commit/PR.  But I think it makes sense to have since apply to all boots, or else we are just knowingly disregarding the users request.

Reply:
The only solution I can think of for this now that keeps the "auto since" behaviour is to ignore `since`/`until` in `add_journal()` if the `boot` kwarg is used - this needs to be documented in the method docstring.

Reply:
> I think it makes sense to have since apply to all boots, or else we are just knowingly disregarding the users request.

We already do that for `messages*` etc. - we're somewhat hamstrung here, in that the request to *always* include the current and last boot came directly from support teams, so it will be seen as a regression if `--since` starts overriding that.

Reply:
I think it can be explained easily enough, but I'm going to reopen #1848 for further discussion on since applying to all journals.  

For now, I'll have this commit/PR just do since for the normal journal and leave this/last without it.

Reply:
I agree that we should make this more consistent - the catch is that there are specific exceptions (like always having the current/last boot in the Red Hat logs data) that we've historically agreed to that we either need to still honour, or to clearly explain when we're making changes.